packages/manager/apps/dedicated-servers/src/pages/listing/index.tsx

```diff
@@ -16,21 +16,30 @@ import {
   ChangelogButton,
   BaseLayout,
   useFeatureAvailability,
+  useResourcesIcebergV6,
 } from '@ovh-ux/manager-react-components';
 import { OdsTabs, OdsTab } from '@ovhcloud/ods-components/react';
 import { useTranslation } from 'react-i18next';
 import { CHANGELOG_LINKS } from '@/data/constants/changelogLinks';
 import { urls } from '@/routes/routes.constant';
+import { Cluster } from '@/data/types/cluster.type';
 
 export default function Layout() {
   const { t } = useTranslation('dedicated-servers');
+  const [hasCluster, setHasCluster] = useState(false);
   const location = useLocation();
   const { shell } = useContext(ShellContext);
   const matches = useMatches();
   const { trackCurrentPage } = useOvhTracking();
   const { data: features, isSuccess } = useFeatureAvailability([
     'dedicated-server:cluster',
   ]);
+  const { flattenData, isSuccess: isSuccessCluster } = useResourcesIcebergV6<
+    Cluster[]
+  >({
+    route: `/dedicated/cluster`,
+    queryKey: ['dedicated-servers', `/dedicated/cluster`],
+  });
```

Comment:
Can you create a hook for this under hooks folder and also get Query key and Query function from api folder ?